tests/bofire/strategies/test_doe.py

```diff
@@ -462,14 +462,27 @@ def calc_volume_content(A, B, T, W, W_T):
         ipopt_options={"maxiter": 500},
     )
     strategy = DoEStrategy(data_model=data_model)
-    doe = strategy.ask(candidate_count=n_experiments, raise_validation_error=False)
-    doe["SC"] = calc_solid_content(*[doe[col] for col in ["A", "B", "T", "W", "W_T"]])
-    doe["VC"] = calc_volume_content(*[doe[col] for col in ["A", "B", "T", "W", "W_T"]])
-    doe["T_calc"] = 0.0182 - 0.03704 * doe["VC"]
-    doe["T_conc"] = doe["T"] / (doe["T"] + doe["W_T"])
-
-    assert np.allclose(doe["T_conc"], 0.03)
-    assert all((doe["VC"] > 0.299) & (doe["VC"] < 0.45))
+
+    # either get validation error or check asserts
+    for _ in range(3):

```

Comment:
Can you document the reason for the retries and explain why it is ok to do that and the functionality is not broken?